src/util/clip.js

```diff
@@ -19,12 +21,112 @@ export function clip(val, low, high) {
  *
  * @param  {Object} point The point to clip
  * @param  {Object} box   The bounding box to clip to.
- * @returns {Object}       The clipped point.
+ * @returns {void}
  */
 export function clipToBox(point, box) {
-  // Clip an {x, y} point to a box of size {width, height}
-  point.x = clip(point.x, 0, box.width);
-  point.y = clip(point.y, 0, box.height);
+  // Clip an {x, y} point to a box {top, left, width, height}
+  const left = box.left || 0;
+  const top = box.top || 0;
+
+  point.x = clip(point.x, left, left + box.width);
+  point.y = clip(point.y, top, top + box.height);
+}
+
+/**
+ * Returns a new bounding box of rotated text box, relative to the pixel
+ * coordinate system. It will get the coordinate of the 4 points of the rotated
+ * text box and calculate the AABB (axis-aligned bounding box - lower and upper
+ * boundaries for `x` and `y` axes).
+ *
+ * @param {HTMLElement} element The element to manipulate pixel positioning
+ * @param {Object} box - `{ left, top, width, height }` in canvas coordinates
+ * @returns {Object} - `{ minX, minY, maxX, maxY }` boundaries of the box
+ */
+const getBoxPixelBoundaries = (element, box) => {
+  const toPixel = point => external.cornerstone.canvasToPixel(element, point);
+  const { top, left, width, height } = box;
+  const topLeft = toPixel({ x: left, y: top });
+  const topRight = toPixel({ x: left + width, y: top });
+  const bottomLeft = toPixel({ x: left, y: top + height });
+  const bottomRight = toPixel({ x: left + width, y: top + height });
+  const points = [topLeft, topRight, bottomLeft, bottomRight];
+  const xArray = points.map(p => p.x);
+  const yArray = points.map(p => p.y);
+
+  return {
+    minX: Math.min(...xArray),
+    minY: Math.min(...yArray),
+    maxX: Math.max(...xArray),
+    maxY: Math.max(...yArray),
+  };
+};
+
+/**
+ * Reposition a box point coordinates in the given axis' upper/lower limits
+ *
+ * @param {Object} point `{ x, y }` The coordinate point of the box
+ * @param {string} axis The axis to be manipulated: `x` or `y`
+ * @param {number} boxMin The box position's lower value on axis
+ * @param {number} boxMax The box position's upper value on axis
+ * @param {*} lowerLimit The lower limit of allowed box position on axis
+ * @param {*} upperLimit The upper limit of allowed box position on axis
+ * @returns {void}
+ */
+const clipBoxOnAxis = (point, axis, boxMin, boxMax, lowerLimit, upperLimit) => {
+  if (upperLimit - lowerLimit < boxMax - boxMin) {
```

Comment:
Did you test it and it doesn't work for you?

We tested it extensively and it does work as expected.

Please let me know what is the issue with the implementation because I couldn't figure out.